functions/go/upsert-resource/main.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/upsert-resource/generated"
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/upsert-resource/upsertresource"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/kio/filters"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+//nolint
+func main() {
+	resourceList := &framework.ResourceList{}
+	resourceList.FunctionConfig = map[string]interface{}{}
+
+	cmd := framework.Command(resourceList, func() error {
+		ur, err := getUpsertResource(resourceList.FunctionConfig)
+		if err != nil {
+			return fmt.Errorf("failed to parse function config: %w", err)
+		}
+		resourceList.Items, err = ur.Filter(resourceList.Items)
+		if err != nil {
+			return fmt.Errorf("failed to upsert resource: %w", err)
+		}
+		return nil
+	})
+
+	cmd.Short = generated.UpsertResourceShort
+	cmd.Long = generated.UpsertResourceLong
+	cmd.Example = generated.UpsertResourceExamples
+
+	if err := cmd.Execute(); err != nil {
+		fmt.Println(err)
```

Comment:
Don't print error to stdout. Instead you can do it in stderr.

Reply:
Updated.